src/rocksdb.rs

```diff
@@ -2340,6 +2341,28 @@ impl Drop for Cache {
     }
 }
 
+pub struct MemoryAllocator {
+    pub inner: *mut DBMemoryAllocator,
+}
+
+impl MemoryAllocator {
+    pub fn new_jemalloc_memory_allocator() -> MemoryAllocator {
```

Comment:
make this method enable only when jemalloc feature is enabled. something like: https://github.com/tikv/tikv/blob/master/components/tikv_alloc/src/lib.rs#L81

the jemalloc feature is defined here: https://github.com/pingcap/rust-rocksdb/blob/master/Cargo.toml#L19

Reply:
Should handle error and return `Result<MemoryAllocator, String>`.

Reply:
@yiwu-arbug rocksdb itself will check if jemalloc is enabled. If it's not the NewJemallocAllocator will assign a nullptr to allocator, which is default value of LRUOptions.

Reply:
so new_jemalloc_memory_allocator() won't fail, I think.

Reply:
`NewJemallocAllocator` can fail when either jemalloc is not linked or jemalloc return error when rocksdb try to create jemalloc hooks. Anyway since underlying c++ method returns a status, for completeness we should have error handling on rust side.

Reply:
@yiwu-arbug You are right.
By it won't fail I mean the memory allocation will always success.